tutorial.md

```diff
@@ -0,0 +1,225 @@
+# Title
+
+Automatic Code Generation with SymPy
+
+# Tutorial Topic
+
+Computational Science and Numerical Techniques
+
+# Student’s Python Knowledge Level
+
+Intermediate
+
+# Please provide a detailed abstract of your tutorial:
+
+This tutorial will introduce code generation concepts using the SymPy library.
+SymPy is a pure Python library for symbolic mathematics. Code generation refers
+to the act of converting a SymPy symbolic expression into equivalent code in
+some language. This allows one to use SymPy to symbolically model a problem,
+and generate fast numerical code for specific platforms that executes that
+model. This is a powerful tool that is useful to scientists in many domains.
+Code generation allows users to speed up existing code, to deal only with the
+high level mathematics of a problem, avoids mathematical errors and typos,
+makes it possible to deal with expressions that would otherwise be too large to
+write by hand, and opens possibilities to perform automatic mathematical
+optimizations of expressions.
+
+SymPy supports generating code for C, C++, Fortran, Matlab/Octave, Python,
+Cython, Julia, Javascript, LLVM, Rust, Haskell, Mathematica, Tensorflow, and
+Theano, and can easily be extended to other languages. SymPy’s code generation
+is used by libraries such as PyDy, pyodesys, sympybotics, pycalphad, and many
+other programs.
+
+## Learning objectives
+
+Attendees will be able to:
+- write SymPy expressions describing mathematical functions and identify the
+  function arguments and outputs.
+- use the SymPy code printers to transform SymPy expressions representing
+  common domain specific functions into multiple output languages.
+- use the SymPy code generation routines to output compilable C code and use
+  Cython to access these functions in Python.
+- generate custom vectorized functions with the three SymPy functions:
+  lambdify, ufuncify, and autowrap.
+- create both custom code printers that make use of specialized C libraries and
+  common subexpression elimination (CSE).
+- subclass the core SymPy printers and create a printer for a custom language.
```

Comment:
This is probably a poor way of putting it here. I'm not sure if we were thinking of any language. We should show how to support anything not supported by default, whether that be a whole language, a library, or just some changing the way something that is already there works.